Test/AwsCommonRuntimeKitTests/mqtt/Mqtt5ClientTests.swift

```diff
@@ -6,6 +6,10 @@ import Foundation
 import AwsCMqtt
 @testable import AwsCommonRuntimeKit
 
+enum TimeoutError: Error {
```

Comment:
Per discussion, feel like there's hackiness/weirdness related to timeouts and async that shouldn't be necessary in a modern language and test framework

Reply:
I've dug some more into this and we could use an `XCTestExpectation` and wrap the async functions within a task that calls `fulfill` on the expectation when it completes and times out when it doesn't but the underlying problem that's being hacked around is that you can't cancel the task once it's awaiting something from native to complete since native doesn't understand the idea of an async task being cancelled. If we wait for the expectation and it times out, it will fail the test but until native completes the continuation, the individual test will continue to run until the test-wide timeout closes everything down. 

I'm open to changing to a hacky way of using expectations in which a timed out expectation results in us manually closing the client but that's not really different from what we're doing now. (and might result in something more complicated to use across all operation calls) At this point, the current implementation is running the test and failing appropriately when it's breaking in the way we expect it to. Unsure it's worth the time to figure out a cleaner way atm but we could revisit this as a later task.